numpy/core/src/multiarray/ctors.c

```diff
@@ -3564,14 +3564,35 @@ PyArray_FromIter(PyObject *obj, PyArray_Descr *dtype, npy_intp count)
 {
     PyObject *value;
     PyObject *iter = PyObject_GetIter(obj);
+    PyObject *iter2, *value2;
     PyArrayObject *ret = NULL;
-    npy_intp i, elsize, elcount;
+    npy_intp i, j, elsize, elcount[2], ndim;
     char *item, *new_data;
 
     if (iter == NULL) {
         goto done;
     }
-    elcount = (count < 0) ? 0 : count;
+    value = PyIter_Next(iter);
+    if (PyErr_Occurred()) {
+        goto done;
+    }
+    elcount[0] = (count < 0) ? 0 : count;
+    if (PySequence_Check(value) && (dtype->fields == Py_None)) {
+        elcount[1] = PySequence_Length(value);
+        if (elcount[1] < 1) {
```

Comment:
Is an empty sequence valid in this context?  I'm not clear on why it should reduce the number of dimensions.